File: spring-cloud-routing-and-filtering/book/src/main/scala/com/github/jccode/springcloud/routingfiltering/BookApplication.scala
package com.github.jccode.springcloud.routingfiltering

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.web.bind.annotation.{RequestMapping, RestController}

@RestController
@SpringBootApplication
class BookApplication {

  @RequestMapping(Array("available")) def available = "Spring in Action"

  @RequestMapping(Array("checked-out")) def checkedOut = "Spring Boot in Action"

}

object BookApplication {
  def main(args: Array[String]): Unit = {
    SpringApplication run classOf[BookApplication]
  }
}

File: spring-cloud-routing-and-filtering/book/src/main/java/com/github/jccode/springcloud/routingfiltering/BookApplication.java
package com.github.jccode.springcloud.routingfiltering;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@SpringBootApplication
public class BookApplication {

    public static void main(String[] args) {
        SpringApplication.run(BookApplication.class, args);
    }

    @RequestMapping("available")
    public String available() {
        return "Spring in Action";
    }

    @RequestMapping("checked-out")
    public String checkedOut() {
        return "Spring Boot in Action";
    }
}
